Add support for the ClaimsParameter to support Twitch OIDC as IdP

vallahaye opened the pull request:
@bnfinet I'm done with the implementation. I tested it locally and it worked as expected (I could login, logout and validate sessions issued with Twitch as IdP).

In the end, I opted for a more generic approach and decided to make the claims configuration available to each provider (in the same way the `code_challenge` parameter is handled).

I would like to have your opinion before adding unit tests.

Also, should I update the `CHANGELOG` as suggested in the [README](https://github.com/vouch/vouch-proxy#contributing-to-vouch-proxy-by-submitting-a-pull-request) ?

Happily waiting for your review :)

#414

bnfinet commented:
@vallahaye thanks for the fine contribution.  That looks straight forward.

Yes thanks, please do update the CHANGELOG.  And could you add `oauth.claims` to `config/config.yml_example` with an explanation and link to #414 and to [the spec](https://openid.net/specs/openid-connect-core-1_0.html#ClaimsParameter).

The `pointer-to-pointer` allows the struct to be uninitialized as `nil`, is that correct?  Otherwise it'd be "zero-valued" as a non-empty `&{map[string]*string`} ...` interface? 

Cheers!

vallahaye commented:
@bnfinet I updated the CHANGELOG and the example configuration. I also added a test to check the `claims` parameter value when starting Vouch with a set of OAuth claims.

> The pointer-to-pointer allows the struct to be uninitialized as nil, is that correct? Otherwise it'd be "zero-valued" as a non-nil &{map[string]*string} ...` interface?

Exactly, a single pointer would initialize the field with a pointer to a zero-value struct, even with the `,omitempty` tag set.

bnfinet commented:
@vallahaye thanks so much for adding `oauth.claims` and Twitch support to VP!

If you're ever in the US Pacific Northwest (someday we all hope) please look me up, I'd love to buy you a beverage.
